pkg/cidata/cidata.TEMPLATE.d/lima.env

```diff
@@ -4,6 +4,10 @@ LIMA_CIDATA_MOUNTS={{ len .Mounts }}
 {{- range $i, $val := .Mounts}}
 LIMA_CIDATA_MOUNTS_{{$i}}_MOUNTPOINT={{$val}}
 {{- end}}
+LIMA_CIDATA_DNS={{ len .DNSAddresses }}
```

Comment:
nit: LIMA_CIDATA_DNS_COUNT might be less confusing

Reply:
Sure. I copied the naming scheme from `LIMA_CIDATA_MOUNTS`, so I will append `_COUNT` there as well for consistency.

Reply:
Since the `DNS` variables are gone, I'm not going to rename the `MOUNTS` variables in this PR.